src/resources/MachineLearning/ModelListing/ModelListing.ts

```diff
@@ -0,0 +1,14 @@
+import API from '../../../APICore.js';
+import Resource from '../../Resource.js';
+import {MLListingModel} from './ModelListingConfiguration.js';
+
+export default class ModelListing extends Resource {
+    static baseUrl = `/rest/organizations/${API.orgPlaceholder}/machinelearning/configuration/modelslisting`;
+
+    list(engineIds?: string[]) {
+        const requestUrl = engineIds?.length
+            ? `${ModelListing.baseUrl}?${engineIds.map((id) => `engineId=${encodeURI(id)}`).join('&')}`
+            : ModelListing.baseUrl;
+        return this.api.get<MLListingModel[]>(requestUrl);
```

Comment:
The new endpoint isn't paginated?

Reply:
Nope, it returns the list of models as a whole. The pagination is done by frontend. We took the decision because the information returned in this endpoint is an aggregate of a few different service that either support pagination or not.